gateway-provider-ha/src/main/java/org/apache/knox/gateway/ha/dispatch/CommonHaDispatch.java

```diff
@@ -240,4 +270,73 @@ default void shiftActiveURL(boolean userAgentDisabled, Optional<URI> backendURI)
             }
         }
     }
+
```

Comment:
It is super confusing to have so much code in an interface.
Maybe only so for old eyes like mine though.
Based on the default modifier being in this class previously, I assume that you are not introducing it here for the first time. I'll think about it a bit more further into the review where multiple inheritance starts to rear its head. :)